src/port1.0/portconfigure.tcl

```diff
@@ -77,14 +88,24 @@ proc portconfigure::should_add_stdlib {} {
     return [expr {$has_stdlib && $is_clang}]
 }
 proc portconfigure::should_add_cxx_abi {} {
-    set is_oldos [expr {[option os.platform] eq "darwin" && [option os.major] < 10}]
-    set is_mp_gcc [string match *g++-mp-* [option configure.cxx]]
-    return [expr {$is_oldos && $is_mp_gcc}]
+    # prior to OS X Mavericks, libstdc++ was the default C++ runtime, so
+    #    assume MacPorts libstdc++ must be ABI compatible with system libstdc++
+    # for OS X Mavericks and above, users must select libstdc++, so
+    #    assume they want default ABI compatibility
+    # see https://gcc.gnu.org/onlinedocs/gcc-5.2.0/libstdc++/manual/manual/using_dual_abi.html
+    return [expr {
+                  [option configure.cxx_stdlib] ne ""          &&
+                  [option configure.cxx_stdlib] ne "libc++"    &&
+                  [option compiler.cxx_standard] >= 2011 &&
+                  [option os.platform] eq "darwin"       &&
+                  [option os.major] < 13
+              }]
 }
 proc portconfigure::construct_cxxflags {flags} {
     if {[portconfigure::should_add_stdlib]} {
         lappend flags -stdlib=[option configure.cxx_stdlib]
-    } elseif {[portconfigure::should_add_cxx_abi]} {
+    }
+    if {[portconfigure::should_add_cxx_abi]} {
         lappend flags -D_GLIBCXX_USE_CXX11_ABI=0
```

Comment:
Shouldn't this macro only be added if using macports-libstdc++?

Reply:
This should be fixed in b5d6ca1c12af180eb3667a45abd6a32934909268.